includes/db.inc

```diff
@@ -35,9 +35,14 @@ function islandora_usage_stats_pid_id($pid) {
  */
 function islandora_usage_stats_pid_to_db($pid) {
   $object = islandora_object_load($pid);
-  return db_insert('islandora_usage_stats_objects')
-    ->fields(array('pid' => $pid, 'label' => $object->label))
-    ->execute();
+  if ($object) {
+    return db_insert('islandora_usage_stats_objects')
+      ->fields(array('pid' => $pid, 'label' => $object->label))
```

Comment:
That would certainly be a lot simpler, but its sweeping the fact that the object doesn't exist under the rug. Handling it with exceptions feels like the right thing to do, even though I like the elegance and ease of your solution.